### PR TITLE
Workout: List & Start

### PR DESCRIPTION
**Pull request by awave1**
Adds ability to list workouts and start doing routines

### CODE REVIEW

File: app/src/main/java/com/group15/fitnesstracker/db/WorkoutExercises.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import androidx.room.ForeignKey
+
+@Entity (
+        primaryKeys = ["workoutId", "exerciseId"],
+        foreignKeys = [
+            ForeignKey(entity = Workout::class, parentColumns = ["workoutId"], childColumns = ["workoutId"]),
+            ForeignKey(entity = SetExercise::class, parentColumns = ["exerciseId"], childColumns = ["exerciseId"])
+        ]
+)
+data class WorkoutExercises (
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/WorkoutExercises.kt:14]: in order to omit redundancy, had to include this entity

File: app/src/main/java/com/group15/fitnesstracker/dashboard/workout/workoutProgress/WorkoutProgressActivity.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        presenter = WorkoutProgressPresenter(this, this)
+        presenter.loadExercises(workoutId)
+
+        adapter = WorkoutProgressAdapter(presenter as WorkoutProgressPresenter)
+
+        exerciseList.layoutManager = LinearLayoutManager(this)
+        exerciseList.adapter = adapter
+
+        finishWorkout.setOnClickListener {
+            val isFinished = adapter.items.filterIndexed { index, _ ->
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/dashboard/workout/workoutProgress/WorkoutProgressActivity.kt:32]: `isFinished` is true if all sets in exercise are complete

**Comment by dartuso** [comment]
FATAL ERROR

`     Caused by: java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.Long
        at android.app.SharedPreferencesImpl.getLong(SharedPreferencesImpl.java:279)
        at com.group15.fitnesstracker.MainActivity.onCreate(MainActivity.kt:23)
        at android.app.Activity.performCreate(Activity.java:7183)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1220)`

**Comment by dartuso** [comment]
W/System.err: io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | Room cannot verify the data integrity. Looks like you've changed schema but forgot to update the version number. You can simply fix this by increasing the version number.
        at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
        at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
        at io.reactivex.internal.operators.maybe.MaybeCallbackObserver.onError(MaybeCallbackObserver.java:82)
        at io.reactivex.internal.operators.maybe.MaybeObserveOn$ObserveOnMaybeObserver.run(MaybeObserveOn.java:98)
        at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:119)
        at android.os.Handler.handleCallback(Handler.java:789)
        at android.os.Handler.dispatchMessage(Handler.java:98)
W/System.err:     at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6938)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
    Caused by: java.lang.IllegalStateException: Room cannot verify the data integrity. Looks like you've changed schema but forgot to update the version number. You can simply fix this by increasing the version number.
        at androidx.room.RoomOpenHelper.checkIdentity(RoomOpenHelper.java:139)
        at androidx.room.RoomOpenHelper.onOpen(RoomOpenHelper.java:119)
        at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.onOpen(FrameworkSQLiteOpenHelper.java:151)
        at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:310)
        at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:194)
        at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.getWritableSupportDatabase(FrameworkSQLiteOpenHelper.java:96)
        at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper.getWritableDatabase(FrameworkSQLiteOpenHelper.java:54)
        at androidx.room.RoomDatabase.query(RoomDatabase.java:251)
W/System.err:     at androidx.room.util.DBUtil.query(DBUtil.java:54)
        at com.group15.fitnesstracker.db.dao.WorkoutDao_Impl$7.call(WorkoutDao_Impl.java:157)
        at com.group15.fitnesstracker.db.dao.WorkoutDao_Impl$7.call(WorkoutDao_Impl.java:154)
        at io.reactivex.internal.operators.maybe.MaybeFromCallable.subscribeActual(MaybeFromCallable.java:46)
        at io.reactivex.Maybe.subscribe(Maybe.java:4154)
        at io.reactivex.internal.operators.maybe.MaybeSubscribeOn$SubscribeTask.run(MaybeSubscribeOn.java:54)
        at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
        at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
W/System.err:     at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:764)

**Pull request merged**